Source/Core/AudioCommon/XAudio2_7Stream.cpp

```diff
@@ -184,9 +224,29 @@ bool XAudio2_7::Start()
 	m_mastering_voice->SetVolume(m_volume);
 
 	m_voice_context = std::unique_ptr<StreamingVoiceContext2_7>
-		(new StreamingVoiceContext2_7(m_xaudio2.get(), m_mixer.get(), m_sound_sync_event));
+		(new StreamingVoiceContext2_7(m_xaudio2.get(), GetMixer(), SupportSurroundOutput(), !m_enablesoundloop));
 
-	return true;
+	return SoundStream::Start();
+}
+
+void XAudio2_7::InitializeSoundLoop()
+{
+
+}
+u32 XAudio2_7::SamplesNeeded()
+{
+	return m_voice_context->BufferReady() ? SOUND_FRAME_SIZE : 0;
+}
+
+void XAudio2_7::WriteSamples(s16 *src, u32 numsamples)
+{
+	m_voice_context->WriteFrame(src, numsamples);
+}
+
+bool XAudio2_7::SupportSurroundOutput()
+{
+	bool surround_capable = SConfig::GetInstance().bDPL2Decoder;
+	return surround_capable;
```

Comment:
+/-0 on that. ```bDPL2Decoder``` doesn't necessarily say that, altho the method name does (somewhat).